prow/tide/tide.go

```diff
@@ -123,13 +123,82 @@ func pickSmallestPassingNumber(prs []pullRequest) (bool, pullRequest) {
 	return smallestNumber > -1, smallestPR
 }
 
+// accumulateBatch returns a list of PRs that can be merged after passing batch
+// testing, if any exist. It also returns whether or not a batch is currently
+// running.
+func accumulateBatch(presubmits []string, prs []pullRequest, pjs []kube.ProwJob) ([]pullRequest, bool) {
+	prNums := make(map[int]pullRequest)
+	for _, pr := range prs {
+		prNums[int(pr.Number)] = pr
+	}
+	type accState struct {
+		prs       []pullRequest
+		jobStates map[string]simpleState
+		// Are the pull requests in the ref still acceptable?
+		validPulls bool
+	}
+	states := make(map[string]*accState)
+	for _, pj := range pjs {
+		if pj.Spec.Type != kube.BatchJob {
+			continue
+		}
+		// If any batch job is pending, return now.
+		if toSimpleState(pj.Status.State) == pendingState {
+			return nil, true
```

Comment:
Why not just continue? Can you guarantee that when a batch result succeeds, there will be no pending batch jobs?

Reply:
> Can you guarantee that when a batch result succeeds, there will be no pending batch jobs? (i.e., of the next batch)

Tide decides when to run the next batch. If there is a successful batch result and no pending batch jobs then we will merge the PRs. We shouldn't be able to get into the state that there is a successful result + pending jobs, because then it means we didn't merge the PRs.